resources/js/components/utils/lang.ts

```diff
@@ -0,0 +1,12 @@
+import {useTransContext} from "@mbarzda/solid-i18next";
+import {TOptions} from "i18next";
+
+/**
+ * A wrapper for useTransContext with the basic overload options, and with better
+ * types for usage as TSX attributes (no null returned).
+ */
+export function getLangFunc() {
+  const [t] = useTransContext();
+  return (key: string | string[], options?: TOptions) =>
+    options ? t(key, options) : t(key);
+}
```

Comment:
a co by się stało, jakby zwrócić po prostu funkcję t? Ten check na options robi coś konkretnego?

Reply:
Wtedy wszystkie te piękne overloady dalej by żyły i dostawałbyś czasem teoretycznie null. A tak to korzystam tylko z dwóch overloadów i nulla nie ma bo oba zwracają string.